docs/EXTENDED-ARCHIVES.md

```diff
@@ -0,0 +1,428 @@
+
+*Some examples of devices are: Beaglebone Black, Beaglebone Green, Intel
+Edison, and Raspberry Pi 3.*
+
+### Client
+
+A client represents the program making use of the information about the other
+entities.
+
+*At the moment of this writing, the only existent client is the Etcher application.*
+
+File format
+-----------
+
+The file format should have the following characteristics:
+
+- Streamability: The extended archive allows the client to consume all its
+  information in a streaming fashion
+
+- Compression: The extended archive should be compressed in order to save
+  network resources
+
+We base this format on the [`tar`][tar] archive format, which has the following
+benefits:
+
+- It's a nearly universal archive format (easy to find robust implementations)
+
+- It allows us to easily impose a file order, which greatly helps with
+  streamability. This is not trivial with `zip`, for instance
+
+- It precedes each file with a header construct, making it streaming friendly
+
+The [`tar`][tar] archive is then compressed with the [`xz`][xz] data
+compression format, which has the following benefits:
+
+- A native, robust, performant, cross-platform and streaming-based
+  decompression library for [node.js][nodejs], called
+  [`lzma-native`][lzma-native], which we battle tested at Etcher
+
+- A way to calculate the precise uncompressed size of the file no matter the
+  size (which is a [known limitation of `gzip` after 4
+  GB](https://github.com/resin-io/etcher/issues/638)), and an operation
+  [`lzma-native`][lzma-native] supports out of the box. This is not possible
+  with `bzip2`
+
+- Decent compression ratio, and compression/decompression time
+
+The resulting file should contain the `.etch` extension.
+
+Archive layout
+--------------
+
+The contents of an extended archive are the followings:
+
+- **REQUIRED** `.meta/manifest.json`
+- `.meta/publisher.svg`
+- `.meta/device.svg`
+- `.meta/image.svg`
+- `.meta/schema.json`
+- `.meta/image.bmap`
+- `.meta/instructions.markdown`
+- `.meta/release-notes.txt`
+- **REQUIRED** `image.<extension>`
+- `.meta/checksum`
+
+### Order
+
+Notice that the order is important. The `.meta/` directory must come before the
+image data, since we need that information before being able to make use of the
+image.
+
+The order of the files inside `.meta/` is not mandatory, however the order
+described above ensures the maximum UX perceived efficiency.
+
+### Graphics
+
+We chose [`svg`][svg] over other image formats since clients can display it on
+any kind of screen resolution without requiring multi-layer files, or multiple
+files workarounds.
+
+Archive files
+-------------
+
+### `.meta/manifest.json`
+
+This is the only required file inside `.meta`, which declares information about
+the image, the target device type, and the publisher. This file may contain the
+following top level properties:
+
+- `publisher`: Information about the image publisher
+- `device`: Information about the image device type
```

Comment:
Sounds reasonable. We will remove this stuff from the first implementation, but we will re-discuss this in the future.